### PR TITLE
Fix some typos in `spec` documents and comments.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**

### CODE REVIEW

**Comment by SethTisue** [comment]
LGTM

**Comment by dongjoon-hyun** [comment]
Thank you, @SethTisue !

File: src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala [hunk 1]
```diff
@@ -222,7 +222,7 @@ trait Interface extends ast.TreeDSL {
         object substIdentsForTrees extends Transformer {
           private def typedIfOrigTyped(to: Tree, origTp: Type): Tree =
             if (origTp == null || origTp == NoType) to
-            // important: only type when actually substing and when original tree was typed
+            // important: only type when actually substring and when original tree was typed
```
Comment (gourlaysama) [inline src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala]: Actually, that's not a typo, "substing" is "substituting" here, not "substring".

Comment (dongjoon-hyun) [inline src/compiler/scala/tools/nsc/transform/patmat/PatternMatching.scala]: Oh, thank you. I'll fix soon.

**Comment by dongjoon-hyun** [comment]
Hi, @gourlaysama . I fixed that as `substituting` .  Thank you, again.

**Comment by dongjoon-hyun** [comment]
Thank you, @retronym !

**Pull request merged**